### DISCUSSION TITLE
select.poll() compatibility with cpython

### DISCUSSION THREAD

**Discussion by pavelrevak** [micropython/micropython#13316] [General]
Hello, I see one difference between cpython and micropython implementation of `select.poll()`, where:

- cpython uses fileno - integer - (filedescriptor),
- micropython uses socket like objects

cpython accepts for register and unregister also objects, but this object must have method `fileno()`
but poll.poll() returns always fileno

my question is how to write compatible code? I have these ideas:
1. stay with inefficient select
2. compare socket first then fileno, or check if returned is integer - it is always ugly

or do you have any other compatible but not ugly idea?

what is the reason for this difference?

thanks

**Comment by jimmo** [comment]
> what is the reason for this difference?

@pavelrevak  MicroPython doesn't have file numbers / file descriptors (except on the unix port, and then only for things that are actually unix files. See https://github.com/micropython/micropython/pull/12138 for some recent history here)

> my question is how to write compatible code?

Yeah, this is an annoying difference. Honestly I'd probably just use `select`. But yes your suggestion of checking if the return type is `int` and mapping the fd back to the object is reasonable.
